plugins/azure/postgresqlserver/diagnosticLoggingEnabled.js

```diff
@@ -0,0 +1,68 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'PostgreSQL Diagnostic Logging Enabled',
+    category: 'PostgreSQL Server',
+    domain: 'Databases',
+    description: 'Ensures diagnostic logging is enabled for PostgreSQL servers.',
+    more_info: 'Enabling diagnostic logging for Azure Database for PostgreSQL servers helps with performance monitoring, troubleshooting, and security optimization.',
+    recommended_action: 'Enable diagnostic logging for all PostgreSQL servers.',
+    link: 'https://learn.microsoft.com/en-us/azure/postgresql/single-server/concepts-server-logs',
+    apis: ['servers:listPostgres', 'diagnosticSettings:listByPostgresServers'],
+    compliance: {
+        hipaa: 'HIPAA requires that a secure audit log record for ' +
+            'write read and delete is created for all ' +
+            'activities in the system.'
+    },
+
+    run: function(cache, settings, callback) {
+        const results = [];
+        const source = {};
+        const locations = helpers.locations(settings.govcloud);
+
+        async.each(locations.servers, (location, rcb) => {
+
+            const servers = helpers.addSource(cache, source,
+                ['servers', 'listPostgres', location]);
+
+            if (!servers) return rcb();
+
+            if (servers.err || !servers.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query for PostgreSQL Servers: ' + helpers.addError(servers), location);
+                return rcb();
+            }
+
+            if (!servers.data.length) {
+                helpers.addResult(results, 0, 'No existing PostgreSQL Servers found', location);
+                return rcb();
+            }
+
+            for (let server of servers.data) {
+                if (!server.id) continue;
+
+                var diagnosticSettings = helpers.addSource(cache, source, 
+                    ['diagnosticSettings', 'listByPostgresServers', location, server.id]);
+ 
+                if (!diagnosticSettings || diagnosticSettings.err || !diagnosticSettings.data) {
+                    helpers.addResult(results, 3, `Unable to query for PostgreSQL Server diagnostic settings: ${helpers.addError(diagnosticSettings)}`,
+                        location, server.id);
+                    continue;
+                }
+
```

Comment:
shouldnt we have check for no existing diagnosticSettings found?.

Reply:
No diagnostic settings found basically means diagnostic logging is not enabled